test/common/stream_info/filter_state_impl_test.cc

```diff
@@ -134,6 +134,81 @@ TEST_F(FilterStateImplTest, WrongTypeGet) {
                             "Data stored under test_name cannot be coerced to specified type");
 }
 
+TEST_F(FilterStateImplTest, IterateThroughListTillEnd) {
+  size_t access_count = 0;
+  size_t destruction_count = 0;
+  filter_state().addToList<TestStoredTypeTracking>(
+      "test_name", std::make_unique<TestStoredTypeTracking>(5, &access_count, &destruction_count));
+  filter_state().addToList<TestStoredTypeTracking>(
+      "test_name", std::make_unique<TestStoredTypeTracking>(5, &access_count, &destruction_count));
+  EXPECT_EQ(0, access_count);
+  EXPECT_EQ(0, destruction_count);
+
+  filter_state().forEachListItem<TestStoredTypeTracking>("test_name",
+                                                         [&](const TestStoredTypeTracking& t) {
+                                                           EXPECT_EQ(5, t.access());
+                                                           return true;
+                                                         });
+
+  EXPECT_EQ(2u, access_count);
+  EXPECT_EQ(0, destruction_count);
+
+  resetFilterState();
+  EXPECT_EQ(2u, access_count);
+  EXPECT_EQ(2u, destruction_count);
+}
+
+TEST_F(FilterStateImplTest, IterateThroughListAndBreak) {
```

Comment:
Can you add a brief comment to each test function explaining what they do?

Reply:
Isn't the long name self explanatory? 
The comment is essentially going to be the same, i.e. 
`// iterate through list object in filter state and test the break feature by returning false..`

Also, just saying: I am following the pattern that this file has (i.e.no comments for any test). A lot of our tests in envoy don't have comments :). So, this alone seems like an exception.

Reply:
In general, I ask for this everywhere and we can improve our test understandability by doing this.